src/gfn/gym/hypergrid.py

```diff
@@ -105,16 +108,7 @@ def make_random_states_tensor(
 
             def update_masks(self) -> None:
                 "Update the masks based on the current states."
-                # The following two lines are for typing only.
-                self.forward_masks = cast(
-                    TT["batch_shape", "n_actions", torch.bool],
-                    self.forward_masks,
-                )
-                self.backward_masks = cast(
-                    TT["batch_shape", "n_actions - 1", torch.bool],
-                    self.backward_masks,
-                )
-
+                self.set_default_typing()
                 self.forward_masks[..., :-1] = self.tensor != env.height - 1
```

Comment:
Do you think we should use set_nonexit_masks for this line ?

Reply:
makes sense, good catch!